### PR TITLE
Bug fix/8237 add unit test which validate post deserlization activities lazy load remain

### PR DESCRIPTION
**Pull request by aditijagtap**

### CODE REVIEW

File: Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs [hunk 1]
```diff
@@ -415,5 +415,43 @@ public void BusinessFlowBrowserActionTest()
...
+            //Put the BF in Test Resource
+            string FileName = TestResources.GetTestResourcesFile(@"Solutions" + Path.DirectorySeparatorChar + "CLI" + Path.DirectorySeparatorChar + "BusinessFlows" + Path.DirectorySeparatorChar + "Flow 1.Ginger.BusinessFlow.xml");
+
+            //Load BF
+            BusinessFlow businessFlow = (BusinessFlow)RS.DeserializeFromFile(FileName);
+
+            Assert.AreEqual(false, businessFlow.LazyLoadFlagForUnitTest);
+
+            int count = businessFlow.Activities.Count();
+
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs]: Validate count as well

File: Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs [hunk 2]
```diff
@@ -415,5 +415,43 @@ public void BusinessFlowBrowserActionTest()
             Assert.AreEqual("NewTab", (from aiv in businessFlow.Activities[0].Acts[0].InputValues where aiv.Param == "GotoURLType" select aiv).FirstOrDefault().Value);
         }
 
+        
+        [TestMethod]
+        public void ActiviyLazyLad()
+        {
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs]: 1. Read XML as string
2. Call DoLazyLoadItems
3. Validate lazy load flag is true
4. Get count and validate
5. Validate lazy load flag is false

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4691078&bid=15684210)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4691078&bid=15684210)

**Pull request merged**